### PR TITLE
Fix test active conditions for beans defined in super class

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes https://github.com/micronaut-projects/micronaut-test/issues/223

### CODE REVIEW

**Comment by dstepanov** [comment]
Includes https://github.com/micronaut-projects/micronaut-test/pull/264 @alvarosanchez

File: test-core/src/main/java/io/micronaut/test/condition/TestActiveCondition.java [hunk 1]
```diff
@@ -63,7 +63,11 @@ public boolean matches(ConditionContext context) {
                             final String rootName = packageName + ".$" + simpleName;
                             return declaringTypeClass.isAssignableFrom(activeSpecClazz) || declaringTypeName.equals(rootName) || declaringTypeName.startsWith(rootName + "$");
                         } else {
-                            return declaringTypeClass.isAssignableFrom(activeSpecClazz) || activeSpecName.equals(declaringTypeName) || declaringTypeName.startsWith(activeSpecName + "$");
+                            final String simpleName = NameUtils.getSimpleName(activeSpecName);
+                            return declaringTypeClass.isAssignableFrom(activeSpecClazz) &&
+                                    definition.getClass().getSimpleName().contains(simpleName) // Check if super class def is part of the current spec
```
Comment (graemerocher) [inline test-core/src/main/java/io/micronaut/test/condition/TestActiveCondition.java:68]: How would this work if you had overlapping names like `SimpleService` and `SimpleServiceHelper` or something?

Comment (dstepanov) [reply test-core/src/main/java/io/micronaut/test/condition/TestActiveCondition.java]: You are right, I have changed the check.

**Comment by asodja** [comment]
Does this fixes also: https://github.com/asodja/micronaut-test/commit/a1fabead8290d6f83a41e17ff814c77807d3133d?

**Comment by dstepanov** [comment]
@asodja It should

**Comment by asodja** [comment]
@dstepanov tested and it works. Awesome, thx!

**Pull request merged**